mutiny-core/src/hermes.rs

```diff
@@ -0,0 +1,315 @@
+                                                        log_error!(logger, "Error decrypting DM: {e}");
+                                                    }
+                                                }
+                                            }
+                                            kind => log_warn!(logger, "Received unexpected note of kind {kind}")
+                                        }
+                                    }
+                                },
+                                Ok(RelayPoolNotification::Message { .. }) => {}, // ignore messages
+                                Ok(RelayPoolNotification::Shutdown) => break, // if we disconnect, we restart to reconnect
+                                Ok(RelayPoolNotification::Stop) => {}, // Currently unused
+                                Ok(RelayPoolNotification::RelayStatus { .. }) => {}, // Currently unused
+                                Err(_) => break, // if we are erroring we should reconnect
+                            }
+                        }
+                        _ = delay_fut => {
+                            if stop.load(Ordering::Relaxed) {
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                if let Err(e) = client.disconnect().await {
+                    log_warn!(logger, "Error disconnecting from relays: {e}");
+                }
+            }
+        });
+
+        Ok(())
+    }
+
+    pub async fn check_available_name(&self, name: String) -> Result<bool, MutinyError> {
+        check_name_request(self.http_client.clone(), &self.base_url, name).await
+    }
+
+    pub async fn reserve_name(&self, name: String) -> Result<(), MutinyError> {
+        // check that we have a name token available
+        let available_tokens = self.blind_auth.available_tokens().await;
+        let available_paid_token =
+            match find_hermes_token(&available_tokens, HERMES_SERVICE_ID, HERMES_PAID_PLAN_ID) {
+                Some(t) => t,
+                None => return Err(MutinyError::NotFound),
+            };
+
+        // check that we have a federation added and get it's id/invite code
+        let federation_identity = match self.get_first_federation().await {
+            Some(f) => f,
+            None => return Err(MutinyError::FederationRequired),
+        };
+
+        // do the unblinding
+        let (nonce, blinding_key) = self
+            .blind_auth
+            .get_unblinded_info_from_token(available_paid_token.clone());
+        let unblinded_sig = unblind_signature(blinding_key, available_paid_token.blind_sig);
+
+        // send the register request
+        let req = RegisterRequest {
+            name: Some(name),
+            pubkey: self.public_key.to_string(),
+            federation_id: federation_identity.federation_id,
+            federation_invite_code: federation_identity.invite_code.to_string(),
```

Comment:
the `federation_id` is a part of the invite code so in theory we don't need to send it